src/Compilers/CSharp/Portable/Symbols/FieldSymbol.cs

```diff
@@ -56,6 +56,8 @@ public TypeSymbol Type
             }
         }
 
+        internal virtual Location ErrorLocation => throw ExceptionUtilities.Unreachable;
```

Comment:
>internal virtual Location ErrorLocation => throw ExceptionUtilities.Unreachable; [](start = 8, length = 80)

In general, we prefer abstract API to virtual APIs with default implementation that throws. Perhaps this API can be moved to specific derived type?